packages/medusa/src/migrations/1674455083104-product_category_product.ts

```diff
@@ -0,0 +1,50 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class productCategoryProduct1674455083104 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `
+        CREATE TABLE "product_category_product" (
+          "id" character varying NOT NULL,
+          "product_category_id" character varying NOT NULL,
+          "product_id" character varying NOT NULL,
+          "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+          "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+
+          CONSTRAINT "PK_product_category_product_id" PRIMARY KEY ("id"),
+          CONSTRAINT "FK_product_category_id" FOREIGN KEY ("product_category_id") REFERENCES product_category("id"),
+          CONSTRAINT "FK_product_id" FOREIGN KEY ("product_id") REFERENCES product("id")
```

Comment:
suggestion(non-blocking): Missed this the first time around. We could add cascade on delete, and no action on update:
```suggestion
          CONSTRAINT "FK_product_category_id" FOREIGN KEY ("product_category_id") REFERENCES product_category("id") ON DELETE CASCADE ON UPDATE NO ACTION,
          CONSTRAINT "FK_product_id" FOREIGN KEY ("product_id") REFERENCES product("id") ON DELETE CASCADE ON UPDATE NO ACTION
```

Reply:
piggyback-suggestion: we should add these definitions into the model layer as well as `ManyToMany(() => ProductCategory, { cascade: ["remove", "soft-remove"] })`